### PR TITLE
DrivAerML: 1M surface points + T_max=50 + lr=8e-4 (standardized rerun)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Your PR #2438 showed that 1M surface points + T_max=50 dramatically improved DrivAerML over 50k points. The result was 36.05% val at T_max=50 (vs 56.91% baseline). However, the comparison was not directly fair because the baseline used `--drivaerml-eval-surface-points 50000` while you used 1M eval points. This PR validates the finding with standardized eval settings and the best LR (lr=8e-4 from baseline vs your lr=5e-4).

## Instructions

Run ONE experiment with standardized settings:

```bash
cd target/icml2026

# Standardized: 1M train points, 50k eval points, best LR, best T_max
python train.py --dataset drivaerml \
  --optimizer adamw --lr 8e-4 --cosine_t_max 50 --no-use-ema \
  --drivaerml-train-surface-points 1000000 \
  --drivaerml-eval-surface-points 50000 \
  --wandb_group taki-drivaerml-1m-standardized --wandb_name taki-drivaerml-1m-tmx50-lr8e4
```

**Note on timing:** With 1M train points, each epoch takes ~80 min (vs 13 min at 50k). The 30-min timeout will fire between epochs, so you will get 1 epoch. This is expected.

**Report** `val_primary/surface_rel_l2_pct`, test metric, epochs completed, and the exact runtime. Target: beat **56.91%** (comparable eval settings). Also report how many train steps per epoch.

## Baseline

| Dataset | Metric | Value | PR |
|---|---|---|---|
| DrivAerML | val_primary/surface_rel_l2_pct | **56.91%** | #2467 (no-EMA, AdamW lr=8e-4, T_max=150, 50k points, 2 epochs) |
| DrivAerML | test_primary/surface_rel_l2_pct | **57.33%** | #2467 |

Baseline command: `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 8e-4 --cosine_t_max 150 --no-use-ema --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000`

**Key question:** With `--drivaerml-eval-surface-points 50000` (same as baseline), does 1M training points + T_max=50 + lr=8e-4 beat 56.91%? Your prior experiment suggests yes — validate it.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT taki:

## Results

### DrivAerML: 1M train surface points + T_max=50 + lr=8e-4 (standardized 50k eval)

| Metric | Baseline (#2467) | This Run | Delta |
|---|---|---|---|
| val_primary/surface_rel_l2_pct | **56.91%** | **34.99%** | **-21.92 pp** |
| test_primary/surface_rel_l2_pct | **57.33%** | **33.48%** | **-23.85 pp** |

### Details

- **Epochs completed:** 1 (timeout fired before epoch 2, as expected)
- **Train steps per epoch:** 3,585
- **Loss (epoch 1):** 0.4637
- **LR at epoch end:** 6.35e-4 (cosine decay from 8e-4, T_max=50)
- **Total runtime:** ~202 min (~120 min training epoch 1, ~80 min test eval)
- **Peak VRAM:** ~54.5 GB (55.7% of 98 GB RTX PRO 6000)
- **W&B run:** [`jmgsymsn`](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/jmgsymsn)

### Command

```bash
CUDA_VISIBLE_DEVICES=0 PYTORCH_ALLOC_CONF=expandable_segments:True python train.py --dataset drivaerml \
  --optimizer adamw --lr 8e-4 --cosine-t-max 50 --no-use-ema \
  --drivaerml-train-surface-points 1000000 \
  --drivaerml-eval-surface-points 50000 \
  --batch-size 1 \
  --wandb-group taki-drivaerml-1m-standardized --wandb-name taki-drivaerml-1m-tmx50-lr8e4-bs1 \
  --agent taki
```

### Notes

- **Batch size reduced to 1** (default batch_size=2 OOM'd at 1M surface points — used ~91 GB before crashing). batch_size=1 fits in ~54.5 GB.
- The 3,585 train steps come from DrivAerML's view system: each of the 394 train cases has ~8.8M surface points, yielding ~9 views per case at the 1M-point window (ceil(8.8M / 1M) = 9), totaling ~3,585 views/steps per epoch.
- With num_workers=0 and PVC-backed data, training is heavily I/O bound — GPU utilization was intermittent (~6-11% when sampled). Data loading dominates wall-clock time.

### What happened

**Strongly positive result.** Training with 1M surface points dramatically beats the 50k-point baseline, even with standardized 50k eval points. The val metric improved from 56.91% → 34.99% and test from 57.33% → 33.48% — both roughly 22-24 percentage points better after just 1 epoch. This validates the prior finding from PR #2438 (which got 36.05% val but with 1M eval points). The improvement is real and not an artifact of eval-point differences.

The key insight is that seeing 1M surface points per training step (vs 50k) gives the model dramatically more spatial coverage of each geometry. Even a single epoch over this richer data is sufficient to outperform the baseline by a wide margin.

### Suggested follow-ups

1. **Multi-epoch run with higher timeout:** Only 1 epoch completed. More epochs could further improve results. A 4-6 hour timeout would allow 2-3 epochs.
2. **Increase num_workers:** Data loading is the bottleneck. Setting num_workers=4 with persistent_workers could significantly reduce wall-clock time by prefetching data while GPU processes.
3. **Gradient accumulation at batch_size=2:** Instead of batch_size=1, could try gradient accumulation to effectively get batch_size=2 without the OOM, potentially improving training stability.
4. **Sweep T_max:** With 3,585 steps/epoch, T_max=50 means the cosine schedule completes ~71 cycles per epoch. A larger T_max (e.g., 100-200) matched to the actual step count might give better LR scheduling.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2493 because Superseded: taki reassigned to #2608 (DrivAerML T_max=15).
